relay-sampling/src/lib.rs

```diff
@@ -315,6 +316,29 @@ impl Display for RuleId {
     }
 }
 
+/// A range of time.
+///
+/// The time range should be applicable between the start and end times, both inclusive
+/// (i.e. [start, end]). There aren't any explicit checks to ensure the end time is equal
+/// to or greater than the start time; the time range isn't valid in such cases.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TimeRange {
+    start: DateTime<Utc>,
+    end: DateTime<Utc>,
```

Comment:
I wonder if we should explicitly support half-open ranges right away, for example:
 - a rule that has no defined start but ends at a certain point.
 - a rule that starts at a point and is valid indefinitely.

As such, we could make the time range non-optional, and build it with two options internally.

Reply:
Done in 98e15ae0137d22404454f5ca0b60ea92fc1b1cf1.